float64[] q
---
float64[] T
